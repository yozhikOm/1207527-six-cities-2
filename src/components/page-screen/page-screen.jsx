import React from 'react';
import {Switch, Route} from 'react-router-dom';
import PropTypes from 'prop-types';
import {MainPage} from '../main-page/main-page.jsx';
import {PropertyDetails} from '../property-details/property-details.jsx';
import {SignIn} from '../sign-in/sign-in.jsx';

import withActiveItem from '../../hocs/with-active-item/with-active-item.js';

const PropertyDetailsWrapped = withActiveItem(PropertyDetails);

const PageScreen = (props) => {

  return <Switch>
    <Route path='/' exact render={() => <MainPage {...props}/>} />

    <Route path='/login' exact render={() =>
      <SignIn isAuthorizationRequired={props.isAuthorizationRequired} authenticateUser={props.authenticateUser} />
    }/>

    <Route path='/offer/:id' exact render={(routeProps) =>
      <PropertyDetailsWrapped {...Object.assign({}, routeProps, props)}/>
    }/>

    <Route
      render={() => (
        <div style={{textAlign: `center`}}>

          <h1>
          404
            <br />
            <small>Page not found</small>
          </h1>
        </div>
      )}
    />
  </Switch>;
};
PageScreen.propTypes = {
  isAuthorizationRequired: PropTypes.bool,
  authenticateUser: PropTypes.func,
};
export {PageScreen};
