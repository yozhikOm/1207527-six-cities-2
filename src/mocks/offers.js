export const offers = [
  {
    "id": 1,
    "title": `Cozy house amsterdam`,
    "type": `House`,
    "price": 650,
    "photos": [
      {
        "src": `img/apartment-01.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.85309666406198],
  },
  {
    "id": 2,
    "title": `Cosy room in spacious apartment`,
    "type": `Room`,
    "price": 45,
    "photos": [
      {
        "src": `img/apartment-02.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.369553943508, 4.85309666406198],
  },
  {
    "id": 3,
    "title": `Elegant private room in stylish house`,
    "type": `Room`,
    "price": 128,
    "photos": [
      {
        "src": `img/room.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.929309666406198],
  },
  {
    "id": 4,
    "title": `Huge Tinyhouse in the green, between sea and city`,
    "type": `House`,
    "price": 1520,
    "photos": [
      {
        "src": `img/apartment-03.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3809553943508, 4.939309666406198],
  }
];
