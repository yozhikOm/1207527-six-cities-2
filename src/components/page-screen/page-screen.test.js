import React from 'react';
import {PageScreen} from './page-screen.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {offers} from '../../mocks/offers.js';

it(`PageScreen компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  const mockUserInfo = {
    id: 2,
    name: `fakeUser`,
    avatarUrl: ``,
    isPro: true
  };

  renderer.render(<PageScreen
    isAuthorizationRequired={true}
    userInfo={mockUserInfo}
    authenticateUser={jest.fn()}
    loadFavorites={jest.fn()}
    favorites={offers}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
