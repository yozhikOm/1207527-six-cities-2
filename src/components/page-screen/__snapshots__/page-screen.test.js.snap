// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageScreen компонент рендерится корректно 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/offer/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    render={[Function]}
  />
</Switch>
`;
